framework/src/st_things/things_stack/utils/things_ping.c

```diff
@@ -372,7 +372,7 @@ static OCStackApplicationResult discover_ping_resource_handler(void *ctx, OCDoHa
 
 		THINGS_LOG_D(TAG, "Device =============> Discovered @ %s:%d", client_response->devAddr.addr, client_response->devAddr.port);
 
-		if (strncmp(ping->addr, client_response->devAddr.addr, strlen(client_response->devAddr.addr)) != 0) {
+		if (strncmp(ping->addr, client_response->devAddr.addr, sizeof(client_response->devAddr.addr)) != 0) {
 			THINGS_LOG_E(TAG, "[Error]: Mismatching address.(ping->addr=%s, recvAddr=%s)", ping->addr, client_response->devAddr.addr);
```

Comment:
using sizeof on uncontrolled module's structure is needs caution.
If you want to use size of on here, it's would better to use the definition "MAX_ADDR_STR_SIZE" from the definition.
otherwise, keep as it is.